### PR TITLE
fix(core): Add an option to enable postgres ssl with default certs

### PR DESCRIPTION
**Pull request by netroy**
CP-809

### CODE REVIEW

**Review by krynble — CHANGES_REQUESTED** [review]
Naming is confusing, I'd expect the other ssl related flags to only work if I enable this one, but it's actually the other way around. Maybe rename to `usePubliclyVerifiableSslCertificate`

It's a bit too long but you get the idea.

File: packages/cli/src/Db.ts [hunk 1]
```diff
@@ -85,19 +85,23 @@ export async function transaction<T>(fn: (entityManager: EntityManager) => Promi
...
-			let ssl: TlsOptions | undefined;
-			if (sslCa !== '' || sslCert !== '' || sslKey !== '' || !sslRejectUnauthorized) {
-				ssl = {
-					ca: sslCa || undefined,
-					cert: sslCert || undefined,
-					key: sslKey || undefined,
-					rejectUnauthorized: sslRejectUnauthorized,
-				};
+			let ssl: TlsOptions | boolean | undefined;
+			if (config.getEnv('database.postgresdb.sslEnabled')) {
```
Comment (krynble) [inline packages/cli/src/Db.ts:89]:
```suggestion
			if (config.getEnv('database.postgresdb.sslDefaultCertificate')) {
```

File: packages/cli/src/config/schema.ts [hunk 2]
```diff
@@ -124,6 +124,12 @@ export const schema = {
 					env: 'DB_POSTGRESDB_SSL_REJECT_UNAUTHORIZED',
 				},
 			},
+			sslEnabled: {
+				doc: 'If SSL should be enabled with default certificates. This overwrites the custom `ssl` section above.',
+				format: 'Boolean',
+				default: false,
+				env: 'DB_POSTGRESDB_SSL_ENABLED',
+			},
```
Comment (krynble) [inline packages/cli/src/config/schema.ts:132]:
```suggestion
			sslDefaultCertificate: {
				doc: 'If SSL should be enabled with default certificates. Do not set this to `true` if you are providing your certificate.',
				format: 'Boolean',
				default: false,
				env: 'DB_POSTGRESDB_SSL_ENABLED',
			},
```

**Review by krynble — CHANGES_REQUESTED** [review]

File: packages/cli/src/Db.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ export function getConnectionOptions(dbType: DatabaseType): ConnectionOptions {
 			const sslKey = config.getEnv('database.postgresdb.ssl.key');
 			const sslRejectUnauthorized = config.getEnv('database.postgresdb.ssl.rejectUnauthorized');
 
-			let ssl: TlsOptions | undefined;
+			let ssl: TlsOptions | boolean | undefined = config.getEnv('database.postgresdb.ssl.enabled');
```
Comment (krynble) [inline packages/cli/src/Db.ts:93]:
```suggestion
			let ssl: boolean | undefined = config.getEnv('database.postgresdb.ssl.enabled');
```

Shouldn't this be just boolean or undefined?

**Review by krynble — DISMISSED** [review]
LGTM - just left a question.

File: packages/cli/src/Db.ts (hunk 3 above)
Comment (netroy) [reply packages/cli/src/Db.ts:93]: it should actually be `TlsOptions | boolean`

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.9.3`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.9.3)

**Pull request merged**